dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Clients/BedrockTextGenerationClient.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading;
+using System.Threading.Tasks;
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+
+/// <summary>
+/// Represents a client for interacting with the text generation through Bedrock.
+/// </summary>
+internal sealed class BedrockTextGenerationClient
+{
+    private readonly string _modelId;
+    private readonly string _modelProvider;
+    private readonly IAmazonBedrockRuntime _bedrockRuntime;
+    private readonly IBedrockTextGenerationIOService _ioTextService;
+    private readonly BedrockClientUtilities _clientUtilities;
+    private Uri? _textGenerationEndpoint;
+    private readonly ILogger _logger;
+
+    /// <summary>
+    /// Builds the client object and registers the model input-output service given the user's passed in model ID.
+    /// </summary>
+    /// <param name="modelId">The model to be used for text generation. </param>
+    /// <param name="bedrockRuntime">The IAmazonBedrockRuntime object to be used for Bedrock runtime actions.</param>
+    /// <param name="loggerFactory">Logger for error output.</param>
+    internal BedrockTextGenerationClient(string modelId, IAmazonBedrockRuntime bedrockRuntime, ILoggerFactory? loggerFactory = null)
+    {
+        var clientService = new BedrockClientIOService();
+        this._modelId = modelId;
+        this._bedrockRuntime = bedrockRuntime;
+        this._ioTextService = clientService.GetTextIOService(modelId);
+        this._modelProvider = clientService.GetModelProviderAndName(modelId).modelProvider;
+        this._clientUtilities = new BedrockClientUtilities();
+        this._logger = loggerFactory?.CreateLogger(this.GetType()) ?? NullLogger.Instance;
+    }
+
+    internal async Task<IReadOnlyList<TextContent>> InvokeBedrockModelAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrWhiteSpace(prompt);
+        var requestBody = this._ioTextService.GetInvokeModelRequestBody(this._modelId, prompt, executionSettings);
+        var invokeRequest = new InvokeModelRequest
+        {
+            ModelId = this._modelId,
+            Accept = "*/*",
+            ContentType = "application/json",
+            Body = new MemoryStream(JsonSerializer.SerializeToUtf8Bytes(requestBody))
```

Comment:
Does the `InvokeModelAsync` method dispose of the memory stream, or should it be disposed of by the code that created it?

Reply:
Thats a valid point, I changed the logic for the text generation so the memorystream is disposed for the requestBody as soon it leaves the `try` block.